File: src/api/export.ts
import { invoke } from "./client";
import type { ExcerptFilter } from "./types";

export const exportCodebookCsv = (path: string) => invoke<void>("export_codebook_csv", { path });
export const exportExcerptsCsv = (path: string, filter: ExcerptFilter) =>
  invoke<void>("export_excerpts_csv", { path, filter });
export const exportActivityCsv = (path: string) => invoke<void>("export_activity_csv", { path });
export const exportProjectJson = (path: string) => invoke<void>("export_project_json", { path });
export const exportCodebookJson = (path: string) => invoke<void>("export_codebook_json", { path });

File: src-tauri/src/commands/export.rs
use std::fs::File;
use std::io::BufWriter;

use misket_core::db::export;
use misket_core::models::ExcerptFilter;
use misket_core::Result;
use tauri::State;

use crate::state::AppState;

#[tauri::command]
pub fn export_codebook_csv(state: State<'_, AppState>, path: String) -> Result<()> {
    state.with_project(|p| export::codebook_csv(&p.conn, BufWriter::new(File::create(&path)?)))
}

#[tauri::command]
pub fn export_excerpts_csv(
    state: State<'_, AppState>,
    path: String,
    filter: ExcerptFilter,
) -> Result<()> {
    state.with_project(|p| {
        export::excerpts_csv(&p.conn, &filter, BufWriter::new(File::create(&path)?))
    })
}

#[tauri::command]
pub fn export_project_json(state: State<'_, AppState>, path: String) -> Result<()> {
    state.with_project(|p| export::project_json(&p.conn, BufWriter::new(File::create(&path)?)))
}

#[tauri::command]
pub fn export_codebook_json(state: State<'_, AppState>, path: String) -> Result<()> {
    state.with_project(|p| export::codebook_json(&p.conn, BufWriter::new(File::create(&path)?)))
}
